types & grammar/ch1.md

```diff
@@ -120,71 +120,71 @@ Outra maneira de pensar sobre tipos do JS é que JS não possui "tipagem forçad
 
 O *valor* `42` possui o tipo `number` intrínseco, e seu *tipo* não pode ser modificado. Um outro valor, como `"42"` com o tipo `string`, pode ser criado *à partir* do valor `42` de tipo `number` através de um processo chamado **coerção (coercion)** (veja o Capítulo 4).
 
-If you use `typeof` against a variable, it's not asking "what's the type of the variable?" as it may seem, since JS variables have no types. Instead, it's asking "what's the type of the value *in* the variable?"
+Se você usa `typeof` em uma variável, não está perguntando "qual o tipo da variável?" como pode parecer, pois variáveis JS não possuem tipos. Pelo contrário, está perguntando "qual é o tipo do valor *na* variável?"
 
 ```js
 var a = 42;
-typeof a; // "number"
+typeof a; // "number" ("número")
 
 a = true;
-typeof a; // "boolean"
+typeof a; // "boolean" ("booleano")
 ```
 
-The `typeof` operator always returns a string. So:
+O operador `typeof` sempre retorna uma string. Assim:
 
 ```js
 typeof typeof 42; // "string"
 ```
 
-The first `typeof 42` returns `"number"`, and `typeof "number"` is `"string"`.
+O primeiro `typeof 42` retorna `"number"`, e `typeof "number"` é `"string"`.
 
 ### `undefined` vs "undeclared"
 
-Variables that have no value *currently*, actually have the `undefined` value. Calling `typeof` against such variables will return `"undefined"`:
+Variáveis que não possuem valores *neste momento*, atualmente possuem o valor `undefined`. Chamar `typeof` nestas variáveis retornará `"undefined"`:
 
 ```js
 var a;
 
-typeof a; // "undefined"
+typeof a; // "undefined" ("indefinido")
 
 var b = 42;
 var c;
 
-// later
+// depois
 b = c;
 
-typeof b; // "undefined"
-typeof c; // "undefined"
+typeof b; // "undefined" ("indefinido")
+typeof c; // "undefined" ("indefinido")
 ```
 
-It's tempting for most developers to think of the word "undefined" and think of it as a synonym for "undeclared." However, in JS, these two concepts are quite different.
+É tentador para muitos desenvolvedores olharem para a palavra "undefined" e pensar nela como sinônimo para "undeclared." Contudo, em JS, esses dois conceitos são bastante diferentes.
 
-An "undefined" variable is one that has been declared in the accessible scope, but *at the moment* has no other value in it. By contrast, an "undeclared" variable is one that has not been formally declared in the accessible scope.
+Uma variável "undefined" é uma que foi declarada no escopo acessível, mas *neste momento* não possui algum outro valor. Por contraste, uma variável "undeclared" é uma que não foi formalmente declarada no escopo acessível.
```

Comment:
Perfeito, vou corrigir :D

Em 11 de junho de 2016 00:28, Cezar Augusto <notifications@github.com>
escreveu:

> In types & grammar/ch1.md
> <https://github.com/cezaraugusto/You-Dont-Know-JS/pull/99#discussion_r66700050>
> :
>
> >
> > -An "undefined" variable is one that has been declared in the accessible scope, but *at the moment* has no other value in it. By contrast, an "undeclared" variable is one that has not been formally declared in the accessible scope.
> > +Uma variável "undefined" é uma que foi declarada no escopo acessível, mas *neste momento* não possui algum outro valor. Por contraste, uma variável "undeclared" é uma que não foi formalmente declarada no escopo acessível.
>
> Uma variável "undefined" é uma *variável* que...
>
> Para um leitor não acostumado, pode gerar confusão: *uma*, se refere a
> variável ou se refere a *uma* *undefined*? Essas coisas.
>
> Por contraste, uma variável "undeclared" é uma *variável* que...
>
> Mesma coisa.
>
> não possui algum *nenhum* outro valor
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cezaraugusto/You-Dont-Know-JS/pull/99/files/36fa648aaeeba6f9eb12389134ca965bc586cff9#r66700050>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ALYYXRlJEcyYGturSsp9JeH96chWxZnfks5qKitQgaJpZM4IzPTM>
> .
>



-- 
Acesse as páginas da nossa empresa e conheça os aplicativos que vão te
ajudar a economizar, a estar bem informado e muito mais! Aguarde
novidades...

[image: Vida App Fanpage] <https://goo.gl/o3SdRs>
[image: Vidaapp Linkedin Company Page] <https://goo.gl/LNmTy7>


Acesse também o nosso site: Vida App <http://vidaapp.com.br>